packages/bee-q/src/components/toast/bq-toast.tsx

```diff
@@ -0,0 +1,223 @@
+  // =====================================
+
+  componentWillLoad() {
+    this.checkPropValues();
+    this.handleTimeChange();
+    this.handleOpenChange();
+  }
+
+  disconnectedCallback() {
+    this.autoDismissDebounce?.cancel();
+  }
+
+  // Listeners
+  // ==============
+
+  @Listen('bqHide')
+  onNotificationHide() {
+    try {
+      toastPortal.removeChild(this.el);
+      // Remove the toast portal from the DOM when there are no more toasts
+      if (toastPortal.querySelector('bq-toast') === null) {
+        toastPortal.remove();
+      }
+    } catch (error) {
+      /**
+       * Skip DOMException error since it could be possible that
+       * in some situations the notification portal is missing
+       */
+      if (error instanceof DOMException) return;
+      throw error;
+    }
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  @Method()
+  async show(): Promise<void> {
+    this.handleShow();
+  }
+
+  @Method()
+  async hide(): Promise<void> {
+    this.handleHide();
+  }
+
+  @Method()
+  async toast() {
+    if (toastPortal.parentElement === null) {
+      document.body.append(toastPortal);
+    }
+
+    toastPortal.appendChild(this.el);
+
+    requestAnimationFrame(() => {
+      this.show();
+    });
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleShow = () => {
+    const ev = this.bqShow.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = true;
+    }
+  };
+
+  private handleHide = () => {
+    const ev = this.bqHide.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = false;
+    }
+  };
+
+  private get iconName() {
+    switch (this.type) {
+      case 'success': {
+        return 'check-circle';
+      }
+      case 'error': {
+        return 'x-circle';
+      }
+      case 'loading': {
+        return 'spinner-gap';
+      }
+      case 'alert': {
+        return 'warning';
+      }
+      case 'info': {
+        return 'info';
+      }
+      default: {
+        return 'info';
+      }
+    }
+  }
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <Host
+        class={{ '!hidden': !this.open }}
+        aria-hidden={!this.open ? 'true' : 'false'}
+        hidden={!this.open ? 'true' : 'false'}
+        role="status"
+      >
+        <output class="bq-toast" part="base">
+          <div class={{ [`bq-toast--icon ${this.type}`]: true, '!hidden': this.hideIcon }} part="icon">
+            <slot name="icon">
+              <bq-icon name={this.iconName} size="24" weight="bold" slot="icon"></bq-icon>
```

Comment:
We should re-export the BqIcon shadow Parts so they can be targetted within the BqToast component 😉 
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/exportparts

```suggestion
              <bq-icon name={this.iconName} size="24" weight="bold" slot="icon" exportparts="icon,svg"></bq-icon>
```

Reply:
Done